compiler/tests/fixtures/demos/uniswap.fe

```diff
@@ -0,0 +1,365 @@
+        idx to: address
+        value: u256
+
+    event Mint:
+        idx sender: address
+        amount0: u256
+        amount1: u256
+
+    event Burn:
+        idx sender: address
+        amount0: u256
+        amount1: u256
+        idx to: address
+
+    event Swap:
+        idx sender: address
+        amount0_in: u256
+        amount1_in: u256
+        amount0_out: u256
+        amount1_out: u256
+        idx to: address
+
+    event Sync:
+        reserve0: u256
+        reserve1: u256
+
+    pub def __init__():
+        self.factory = msg.sender
+
+    pub def factory() -> address:
+        return self.factory
+
+    pub def token0() -> address:
+        return self.token0
+
+    pub def token1() -> address:
+        return self.token1
+
+    def _mint(to: address, value: u256):
+        self.total_supply = self.total_supply + value
+        self.balances[to] = self.balances[to] + value
+        emit Transfer(address(0), to, value)
+
+    def _burn(from: address, value: u256):
+        self.balances[from] = self.balances[from] - value
+        self.total_supply = self.total_supply - value
+        emit Transfer(from, address(0), value)
+
+    def _approve(owner: address, spender: address, value: u256):
+        self.allowances[owner][spender] = value
+        emit Approval(owner, spender, value)
+
+    def _transfer(from: address, to: address, value: u256):
+        self.balances[from] = self.balances[from] - value
+        self.balances[to] = self.balances[to] + value
+        emit Transfer(from, to, value)
+
+    pub def approve(spender: address, value: u256) -> bool:
+        self._approve(msg.sender, spender, value)
+        return true
+
+    pub def transfer(to: address, value: u256) -> bool:
+        self._transfer(msg.sender, to, value)
+        return true
+
+    pub def transferFrom(from: address, to: address, value: u256) -> bool:
+        assert self.allowances[from][msg.sender] >= value
+
+        self.allowances[from][msg.sender] = self.allowances[from][msg.sender] - value
+        self._transfer(from, to, value)
+        return true
+
+    pub def balanceOf(account: address) -> u256:
+        return self.balances[account]
+
+    # TODO: support structs in the public interface (https://github.com/ethereum/fe/issues/282)
+    pub def get_reserves(): # -> ThreeNums:
+#        return ThreeNums(
+#            num1=self.reserve0,
+#            num2=self.reserve1,
+#            num3=self.block_time_stamp
+#        )
+        pass
+
+    # called once by the factory at time of deployment
+    pub def initialize(token0: address, token1: address):
+        assert msg.sender == self.factory, "UniswapV2: FORBIDDEN"
+        self.token0 = token0
+        self.token1 = token1
+
+    # update reserves and, on the first call per block, price accumulators
+    def _update(balance0: u256, balance1: u256, reserve0: u256, reserve1: u256):
+        #assert balance0 <= u112(-1) and balance1 <= u112(-1), "UniswapV2: OVERFLOW"
```

Comment:
I guess `u112(-1)` is a solidity hack to mean `u112.max_value`. But do you know why this uses an `u112`? I guess we could also just initialize up a `u256` with the whatever the max value of a `u112` is and put that in a constant so that these would become: `assert balance0 <= SOME_MAXIMUM and balance1 <= SOME_MAXIMUM, "UniswapV2: OVERFLOW"`

Reply:
`u112` is used for more efficient storage packing of reserves. I think since we're using `u256` instead of `u112` for reserves, we could just delete this assert entirely.